snap/local/start-mysql-router.sh

```diff
@@ -4,8 +4,11 @@
 # Execute mysqlrouter as the non-sudo user: snap-daemon.
 EXTRA_OPTIONS="$(snapctl get mysqlrouter.extra-options)"
 
-exec $SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
-  --regid snap_daemon \
-  -- $SNAP/usr/bin/mysqlrouter \
-  --config $SNAP_DATA/etc/mysqlrouter/mysqlrouter.conf \
-  $(echo "$EXTRA_OPTIONS")
+exec "${SNAP}/usr/bin/setpriv" \
+    --clear-groups \
+    --reuid snap_daemon \
+    --regid snap_daemon \
+    -- \
+    "${SNAP}/usr/bin/mysqlrouter" \
+    --config "${SNAP_DATA}/etc/mysqlrouter/mysqlrouter.conf" \
+    "${EXTRA_OPTIONS}"
```

Comment:
I suspect `$(echo "$EXTRA_OPTIONS")` used here due to wrong way of working with arrays as strings above.

I am not sure how to test this... but I bet it should be:
```
declare -a EXTRA_OPTIONS=()
EXTRA_OPTIONS+="$(snapctl get mysqlrouter.extra-options)"
...
    "${EXTRA_OPTIONS[@]}"
```

Is it really works well now with simple `"${EXTRA_OPTIONS}"`?

Reply:
in fact, it does not. The construction with `echo` indeed look as the best solution.

I've modified the smoke test to not skip `mysqld_exporter` so we can catch that on CI

Reply:
In fact, using echo here is wrong, it will not work with spaces and wired quotations.
The save wait is to use array and dump it using "${arr[@]}".

Reply:
ddc89d0 - using arrays wherever possible